extensions/http-curl/protocols/RESTSender.cpp

```diff
@@ -139,6 +139,12 @@ const C2Payload RESTSender::sendPayload(const std::string url, const Direction d
   }
   bool isOkay = client.submit();
   int64_t respCode = client.getResponseCode();
+  if (400 <= respCode && respCode < 600) {
+    // client or server error
```

Comment:
minor, but instead of resorting to comments, we could extract these conditions to variables and have their name carry the information, maybe something like this?
```suggestion
  const bool clientError = 400 <= respCode && respCode < 500;
  const bool serverError = 500 <= respCode && respCode < 600;
  if (clientError || serverError) {
```